lib/chef/resource/windows_firewall_rule.rb

```diff
@@ -67,11 +67,18 @@ class WindowsFirewallRule < Chef::Resource
         description: "The action of the firewall rule.",
         coerce: proc { |f| f.is_a?(String) ? f.downcase.to_sym : f }
 
-      property :profile, [Symbol, String],
-        default: :any, equal_to: %i{public private domain any notapplicable},
```

Comment:
you should still validate that the array does not contain anything that isn't one of these symbols?

Reply:
How though? I made it work with the below snippet, but is there a simpler approach?
```ruby
        callbacks: {
                      'contains values not in :public, :private :domain, :any or :notapplicable' => lambda {
                        |p| %i{public private domain any notapplicable} & p == p
                      }
                   }
```

Reply:
that may be overly terse.  `lambda { |p| p.all? { |e| %i{public private domain any notapplicable}.include?(e) } }` or something like that which doesn't require realizing those are set operators.